python/bencode.py

```diff
@@ -1,76 +1,72 @@
 #!/usr/bin/python3
 
-def __decode_string(s):
+def __decode_string(b, start=0):
     """
-    >>> __decode_string('3:foo3:bar')
-    {'value': 'foo', 'rest': '3:bar'}
-    >>> __decode_string('0:3:bar')
-    {'value': '', 'rest': '3:bar'}
+    >>> __decode_string(b'3:foo3:bar')
+    {'value': 'foo', 'start': 5}
```

Comment:
Could you add a test case which call this function with `start` parameter?

Reply:
Same for `__decode_integer`, `__decode_list` and `__decode_dict`

Reply:
It's done.